org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -217,10 +221,20 @@ public static boolean isOffsetInRange(int offset, Range range, IDocument documen
 		}
 	}
 
-	public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocument document)
+	public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocument document, char[] completionTriggerChars)
 			throws BadLocationException {
 		Position start = toPosition(offset, document);
 		final var param = new CompletionParams();
+		try {
+			String positionCharacter = document.get(offset-1, 1);
+			if (Chars.contains(completionTriggerChars, positionCharacter.toCharArray()[0])) {
```

Comment:
I think `positionCharacter.charAt(0)` is nicer

Reply:
Maybe `document.get(offset-1, 1).charAt(0)` in the previous line even better.

Reply:
> Maybe document.get(offset-1, 1).charAt(0) in the previous line even better.

I would use `positionCharacter.charAt(0)`, because I need the `positionCharacter` as String for the constructor of CompletionContext